### PR TITLE
MCO-1416: Create OCL e2e Test Suite & Remove Unneeded Helper

### PR DESCRIPTION
**Pull request by isabella-janssen**
**- What I did**
- Moved the OCL test files from the `test/e2e-techpreview` folder to a new `test/e2e-ocl` folder.
- Removed the unneeded `MustHaveFeatureGatesEnabled` function.

**- How to verify it**
Test the Makefile update by running `make test-e2e-ocl`.
```bash
$ make test-e2e-ocl --quiet
Installing go-junit-report
2024/11/21 09:47:57 All required featuregates [OnClusterBuild MachineConfigNodes] present!
=== RUN   TestOnClusterBuildsOnOKD
    helpers_test.go:630: OCP detected, skipping test TestOnClusterBuildsOnOKD
    helpers_test.go:631: 
--- SKIP: TestOnClusterBuildsOnOKD (0.08s)
=== RUN   TestOnClusterBuildsCustomPodBuilder
...
```

**- Description for the changelog**
[MCO-1416](https://issues.redhat.com//browse/MCO-1416): Separate OCL e2e tests from tech preview and remove unneeded helper

### CODE REVIEW

File: test/e2e-ocl/main_test.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	"testing"
+
+	"github.com/openshift/machine-config-operator/test/helpers"
+)
+
+func TestMain(m *testing.M) {
+
+	// Ensure required feature gates are set.
+	// Add any new feature gates to the test here, and remove them as features are GAed.
+	helpers.MustHaveFeatureGatesEnabled("OnClusterBuild", "MachineConfigNodes")
```
Comment (djoshy) [inline test/e2e-ocl/main_test.go:14]: I think we can remove this helper function altogether as we have the featureset specified [in your companion PR](https://github.com/openshift/release/pull/58848/files#diff-f79ad35584926f5ddf68b3b78fe16e1adea24224a90ab826ad448d118854f6f5R224) in openshift/release. I believe we originally added this check in the tech preview suite when feature gate were relatively new and we weren't sure how they should be configured in CI. 

After both PRs merge, and sometime in the not too _distant_ future when the OCL feature gate moves to default, we can finally remove the featureset config for `e2e-gcp-op-ocl` in `openshift/release` as a follow-up PR :smile:

**Comment by umohnani8** [comment]
Changes LGTM

**Comment by djoshy** [comment]
Thanks for the fix-ups! Let's get this in so we can start testing on https://github.com/openshift/release/pull/58848

/lgtm
/approve

Not a critical fix, but since this is a change to MCO side CI, this should not affect payloads
/label acknowledge-critical-fixes-only

**Pull request merged**